harmony/src/main/java/com/frybits/harmony/Harmony.kt

```diff
@@ -285,12 +285,12 @@ private class HarmonyImpl constructor(
         } catch (e: IllegalStateException) {
             _InternalHarmonyLog.e(LOG_TAG, "IllegalStateException while reading data file", e)
             null to emptyMap()
-        } catch (e: JSONException) {
-            _InternalHarmonyLog.e(LOG_TAG, "JSONException while reading data file", e)
-            null to emptyMap()
         } catch (e: IOException) {
             _InternalHarmonyLog.e(LOG_TAG, "IOException occurred while reading json", e)
             null to emptyMap()
+        } catch (e: SerializationException) {
+            _InternalHarmonyLog.e(LOG_TAG, "Unable to parse json string", e)
```

Comment:
In this case would the client receive the default value?

Reply:
This function reads the preferences from the `prefs.data` file, which holds the current state of all preferences. If there is a failure in reading the file, this is equivalent to clearing preferences. Similar logic exists for [`SharedPreferences`](https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/core/java/android/app/SharedPreferencesImpl.java;l=207?q=SharedPreferencesImp).